pkg/apis/admission/types.go

```diff
@@ -19,72 +19,73 @@ package admission
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/kubernetes/pkg/apis/authentication"
 )
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// AdmissionRequest describes an admission request.
-type AdmissionRequest struct {
+// AdmissionReview describes an admission review request/response.
+type AdmissionReview struct {
 	metav1.TypeMeta
 
-	// Spec describes the attributes for the admission request.
-	// Since this admission controller is non-mutating the webhook should avoid setting this in its response to avoid the
-	// cost of deserializing it.
-	Spec AdmissionReviewSpec
+	// Request describes the attributes for the admission request.
+	Request AdmissionRequest
+
+	// Response describes the attributes for the admission response.
+	// +optional
+	Response AdmissionResponse
 }
 
-// AdmissionReviewSpec describes the admission.Attributes for the admission request.
-type AdmissionReviewSpec struct {
+// AdmissionRequest describes the admission.Attributes for the admission request.
+type AdmissionRequest struct {
+	// UID is an identifier for the individual request. It allows us to distinguish instances of requests which are
+	// otherwise identical (parallel requests, requests when earlier requests did not modify etc)
```

Comment:
// It is suitable for correlating log entries between the webhook and apiserver, for either auditing or debugging.

?

Reply:
Done.

Reply:
Is it unique to the webhook request or to the originating user request (would two webhooks get the same uid for calls resulting from a single user request, or would they each get their own unique uid)?

Reply:
They should each get their own, I think--if we did a UID per request then we'd probably also want a nonce or something here.

Reply:
sounds ok, should clarify that in the field doc